docs/sdks/typescript/quickstart.md

```diff
@@ -142,365 +149,513 @@ function App() {
+
+```tsx
+  const [connected, setConnected] = useState<boolean>(false);
+  const [identity, setIdentity] = useState<Identity | null>(null);
+  const [conn, setConn] = useState<DBConnection | null>(null);
+
+  useEffect(() => {
+    const onConnect = (
+      conn: DBConnection,
+      identity: Identity,
+      token: string
+    ) => {
+      setIdentity(identity);
+      setConnected(true);
+      localStorage.setItem('auth_token', token);
+      console.log(
+        'Connected to SpacetimeDB with identity:',
+        identity.toHexString()
+      );
+      conn
+        .subscriptionBuilder()
+        .onApplied(() => {
+          console.log('SDK client cache initialized.');
+        })
+        .subscribe(['SELECT * FROM message', 'SELECT * FROM user']);
+    };
 
-To the body of `App`, add:
+    const onDisconnect = () => {
+      console.log('Disconnected from SpacetimeDB');
+      setConnected(false);
+    };
 
-```typescript
-function userNameOrIdentity(user: User): string {
-  console.log(`Name: ${user.name} `);
-  if (user.name !== null) {
-    return user.name || '';
-  } else {
-    var identityStr = new Identity(user.identity).toHexString();
-    console.log(`Name: ${identityStr} `);
-    return new Identity(user.identity).toHexString().substring(0, 8);
-  }
-}
+    const onConnectError = (_conn: DBConnection, err: Error) => {
+      console.log('Error connecting to SpacetimeDB:', err);
+    };
 
-function setAllMessagesInOrder() {
-  let messages = Array.from(Message.all());
-  messages.sort((a, b) => (a.sent > b.sent ? 1 : a.sent < b.sent ? -1 : 0));
+    setConn(
+      DBConnection.builder()
+        .withUri('ws://localhost:3000')
+        .withModuleName('quickstart-chat')
+        .withToken(localStorage.getItem('auth_token') || '')
+        .onConnect(onConnect)
+        .onDisconnect(onDisconnect)
+        .onConnectError(onConnectError)
+        .build()
+    );
+  }, []);
+```
 
-  let messagesType: MessageType[] = messages.map(message => {
-    let sender_identity = User.findByIdentity(message.sender);
-    let display_name = sender_identity
-      ? userNameOrIdentity(sender_identity)
-      : 'unknown';
+Here we are configuring our SpacetimeDB connection by specifying the server URI, module name, and a few callbacks including the `onConnect` callback. When `onConnect` is called after connecting, we store the connection state, our `Identity`, and our SpacetimeDB credentials in our React state. If there is an error connecting, we print that error to the console as well.
 
-    return {
-      name: display_name,
-      message: message.text,
-    };
-  });
+We are also using `localStorage` to store our SpacetimeDB credentials. This way, we can reconnect to SpacetimeDB with the same `Identity` and token if we refresh the page. The first time we connect, we won't have any credentials stored, so we pass `undefined` to the `withToken` method, this will cause SpacetimeDB to generate new credentials for us.
```

Comment:
```suggestion
We are also using `localStorage` to store our SpacetimeDB credentials. This way, we can reconnect to SpacetimeDB with the same `Identity` and token if we refresh the page. The first time we connect, we won't have any credentials stored, so we pass `undefined` to the `withToken` method. This will cause SpacetimeDB to generate new credentials for us.
```